example_app_generator/generate_stuff.rb

```diff
@@ -84,8 +84,12 @@ def using_source_path(path)
 generate('rspec:install')
 generate('controller wombats index') # plural
 generate('controller welcome index') # singular
-generate('rspec:request wombats')
-generate('integration_test widgets')
+# Generate controllers so that Rails generates routes
+generate('controller logins index --no-request_specs --no-view_specs --no-helper_specs')
+generate('controller signups index --no-request_specs --no-view_specs --no-helper_specs')
+# The generated specs from IntegrationGenerator rely on index routes being present
```

Comment:
Have you seen the [test output](https://travis-ci.org/github/rspec/rspec-rails/builds/719925902) for the commit before that 9f0bc1fb893e9a26361184706a911cc3d0b4a55b?

The IntegrationSpecGenerator / RequestSpecGenerator generate a file with an index_path helper that does not exist when we have not generated the corresponding route yet, causing a failure of the smoke tests. So there was a hidden dependency between generating the controllers and testing the generated request specs. That I think you understood, right?
This was a very surprising dependency by the way. Dependencies in this `generates_stuff` file are quite the pain generally. And I assume that these test failures were what motived @eloyesp to change the behaviour in the previous PR.

When we stick to wombats and widgets, then we would try to overwrite the already generated request spec files from the earlier lines. `generate('controller wombats index')` generates `spec/requests/wombats_spec.rb`. Then `generate('rspec:request wombats')` was **also** trying to generate `spec/requests/wombats_spec.rb`, so a conflict. We could try using the `--force` option to just ignore that file conflict but I thought that then it would be pointless to even use the request spec generator here.

The usage of `--no-request_specs` is necessary so that generating the controller does not change the conflicting file name. The usage `--no-view_specs --no-helper_specs` options I just added to avoid unnecessary increase in testing time.

Reply:
There are controllers generated above, if they are missing a route then ok add some more, but don't remove the old ones.

Reply:
I don't understand what you mean.
Could you please checkout the commit locally, run `rake smoke:app`, see the failures and tell more detailed how you would solve them?

I did not remove anything. I changed the names for using the request spec generation and the integration_test generation.

- `generate('rspec:request wombats')` fails because `generate('controller wombats index')` already has generated the request spec file. This was not an issue before because `generate('rspec:request wombats')` generated `spec/requests/wombats_request_spec` file. And this file worked only because `generate('controller wombats index')` created the correct route.
- `generate('integration_test widgets')` would work, but the file `spec/requests/widgets_spec` will later on be overwritten by `generate('scaffold widget name:string category:string instock:boolean foo_id:integer bar_id:integer --force')`

Reply:
Ok, I pulled down your branch and ran it and the main issue is that the generator now overwrites itself due to the change in defaults, fixed in #2378